ocp.ci: Various fixes and improvements to the OCP pipeline

ldoktor opened the pull request:
Please see the individual commits for details.

Fixes:  #9228
Related to: #9206

gkurz commented:
/test

wainersm commented on ci/openshift-ci/cluster/install_kata.sh:
```diff
@@ -147,6 +147,44 @@ debug_pod() {
+		pods=$(oc get pods -l app="$app" --no-headers=true $namespace | awk '{print $1}' | xargs)
+		[ -n "$pods" ] && break
+		if [ "$SECONDS" -gt "$timeout" ]; then
+			echo "Unable to find any pods for '-l app=\"$app\" in ${SECONDS}s"
```
Not closing the opened `'`

wainersm replied:
hi @ldoktor ,

Unless I am missing a magic feature of shell, I think it needs to increment `$SECONDS`, doesn't it?

wainersm replied:
oh, I learned something new today! From bash's manpage:

```
 SECONDS
              Each  time  this parameter is referenced, it expands to the number of seconds since shell invocation.  If a value is assigned to SECONDS, the value returned upon subsequent references is the number of seconds since the
              assignment plus the value assigned.  The number of seconds at shell invocation and the current time are always determined by querying the system clock.  If SECONDS is unset, it loses its special properties, even if  it
              is subsequently reset.
```

ldoktor replied:
Bash is full of surprises ;-). Anyway I'll fix the missing `'` in v2

wainersm commented on ci/openshift-ci/cluster/install_kata.sh:
```diff
@@ -185,7 +223,4 @@ fi
-	sleep 10
-	oc logs -n kube-system ds/relabel-selinux-daemonset | grep "NSENTER_FINISHED_WITH:" && exit
-done ) || { echo "Selinux relabel failed, check the logs"; exit -1; }
+wait_for_app_pods_message restorecon "NSENTER_FINISHED_WITH:" 120 "kube-system" || echo "Failed to treat selinux, proceeding anyway..."
```
Wondering why not `exit -1` like before (`{ echo "Selinux relabel failed, check the logs"; exit -1; }`)

ldoktor replied:
I'm fine either way, in this way I consider such failure more like a warning and I'd prefer having testing performed with chance of looking into logs for failures rather than not performed tests because of some bad exit code.

wainersm commented on ci/openshift-ci/cluster/deployments/workaround-9206-crio.yaml:
```diff
@@ -0,0 +1,18 @@
+    storage:
+      files:
+      - contents:
+              source: data:text/plain;charset=utf-8;base64,W2NyaW9dCnN0b3JhZ2Vfb3B0aW9uID0gWwoJIm92ZXJsYXkuc2tpcF9tb3VudF9ob21lPXRydWUiLApdCg==
```
Did you consider to have the config file versioned in plain-text then encoded to base64 on runtime?

Like it does to apply SELinux config -> https://github.com/kata-containers/kata-containers/blob/main/ci/openshift-ci/cluster/deployments/machineconfig_selinux.yaml.in#L20C62-L20C81

The advantage of plain-text is that it's reviewable more easily :D

ldoktor replied:
I can do that but it just seemed better, condensed and hopefully nobody will have to modify it ever :-)

wainersm replied:
ok, your call :)

wainersm commented on ci/openshift-ci/cluster/install_kata.sh:
```diff
@@ -147,6 +147,44 @@ debug_pod() {
+	local i
+	SECONDS=0
+	while :; do
+		pods=$(oc get pods -l app="$app" --no-headers=true $namespace | awk '{print $1}' | xargs)
```
No doubt that I have is whether the daemonset is going to schedule N pods at same time or not. In a situation where a node is not ready *yet* then it won't schedule to that node?

gkurz replied:
> No doubt that I have is whether the daemonset is going to schedule N pods at same time or not. In a situation where a node is not ready _yet_ then it won't schedule to that node?

The number of pods doesn't depend on the status of the nodes specifically. It only depends on the selector.
The pods being scheduled on these nodes is another story.

Since the same selector that is used in the daemonset YAML is passed to `oc get pods`, we should expect to get the exact list of pods at some point in time. Not sure it is guaranteed that all pods are created atomically.

Maybe safer to only break when we get the expected number of pods, which can be passed as an argument to this function.

ldoktor replied:
Sounds good, I'll update that

wainersm commented:
hi @ldoktor , I finished my review. Overall looks good. Left only a few comments ;)

ldoktor commented:
Changes:

* Added the missing `'`
* In wait_for_app_pods_message function used an array to store the list of pods and added check for the expected number of pods
* Extracted the fail->warn for selinux relabel change into a separate commit (can be removed)

Not changed:

* Kept the workaround content as base64 as it doesn't seem that important to use a longer format (open to negotiation)

wainersm reviewed (approved):
hi @ldoktor ! Thanks for those improvements and teaching new tricks to this old dog ;)

wainersm commented:
/test

gkurz commented:
/retest-ubuntu-main

ldoktor commented:
Hello @gkurz, do I need to rebase or so? My code should not touch any tested code (it's only used by the osc pipeline)
